src/nidigital/system_tests/test_system_nidigital.py

```diff
@@ -244,3 +240,83 @@ def test_fetch_capture_waveform(multi_instrument_session):
         assert fetched_site == 1
         assert len(fetched_waveform[fetched_site]) == num_samples
 
+
+def test_get_pin_results_pin_information_with_channel_list(multi_instrument_session):
+    create_pin_map(multi_instrument_session)
+
+    fully_qualified_channels = [instr[1] + '/0', instr[0] + '/1']
+    pin_info = multi_instrument_session.channels[fully_qualified_channels].get_pin_results_pin_information()
+
+    pins = [i.pin_name for i in pin_info]
+    sites = [i.site_number for i in pin_info]
+    channels = [i.channel_name for i in pin_info]
+
+    assert pins == ['PinA', 'PinB']
+    assert sites == [1, 0]
+    assert channels == fully_qualified_channels
+
+
+def test_get_pin_results_pin_information_with_pin_list(multi_instrument_session):
+    create_pin_map(multi_instrument_session)
+
+    pin_info = multi_instrument_session.pins['SysPins, site0/DutPins'].get_pin_results_pin_information()
+
+    pins = [i.pin_name for i in pin_info]
+    sites = [i.site_number for i in pin_info]
+    channels = [i.channel_name for i in pin_info]
+
+    assert pins == ['SysPin', 'PinA', 'PinB', 'PinC']
+    assert sites[1:] == [0, 0, 0]
+    assert channels == [instr[0] + '/3', instr[0] + '/0', instr[0] + '/1', instr[0] + '/2']
+
+
+def test_get_pin_results_pin_information_on_unmapped_channels(multi_instrument_session):
+    create_pin_map(multi_instrument_session)
+
+    fully_qualified_channels = [instr[1] + '/11', instr[0] + '/10']
+    pin_info = multi_instrument_session.channels[fully_qualified_channels].get_pin_results_pin_information()
+
+    pins = [i.pin_name for i in pin_info]
+    sites = [i.site_number for i in pin_info]
+    channels = [i.channel_name for i in pin_info]
+
+    assert pins == ['', '']
+    assert sites == [-1, -1]
+    assert channels == fully_qualified_channels
+
+
+def test_get_pin_results_pin_information_on_all_channels(multi_instrument_session):
+    create_pin_map(multi_instrument_session)
+
+    pin_info = multi_instrument_session.get_pin_results_pin_information()
```

Comment:
A lot of these tests call the same `get_pin_results_pin_information` method.
What additional coverage are we getting?

We need to be very careful to scope these tests so they ensure that the Python bindings interact with the driver runtime correctly, but not that the driver runtime behaves as it should. That's the job of NI's internal Automated Test Suite.

Reply:
Even though the public method is `get_pin_results_pin_information`, it internally calls `get_pin_name` and `get_channel_name`.

`test_get_pin_results_pin_information_on_all_channels` is the "standard" test for `get_pin_results_pin_information`
`test_get_pin_results_pin_information_with_channel_list` is testing the `channels` repeated capability
`test_get_pin_results_pin_information_with_pin_list` is testing the `pins` repeated capability
`test_get_pin_results_pin_information_on_unmapped_channels` is verifying handling of -1 returned for pin index.

Reply:
I guess I could rename `test_get_pin_results_pin_information_with_channel_list` as `test_channels_repeated_capability` and `test_get_pin_results_pin_information_with_pin_list` as `test_pins_repeated_capability`